external/vcm/vcm/derived_mapping.py

```diff
@@ -58,6 +58,27 @@ def _data_arrays(self, keys: Iterable[Hashable]):
     def dataset(self, keys: Iterable[Hashable]) -> xr.Dataset:
         return xr.Dataset(self._data_arrays(keys))
 
+    @classmethod
+    def find_all_required_inputs(
+        cls, derived_variables: Iterable[str]
+    ) -> Iterable[str]:
+        # Helper function to find full list of required inputs for a given list
+        # of derived variables. Recurses because some required inputs have their
+        # own required inputs (e.g. pQ's)
+        def _recurse_find_deps(vars, deps):
+            vars_with_deps = [var for var in vars if var in cls.REQUIRED_INPUTS]
+            if len(vars_with_deps) == 0:
+                return
+            else:
+                for var in vars_with_deps:
+                    new_deps = cls.REQUIRED_INPUTS[var]
+                    deps += new_deps
+                _recurse_find_deps(new_deps, deps)
```

Comment:
I think this might be a bug lying in wait.  `new_deps` is set by the last iteration in the loop, so the recursion would miss any new dependencies from previous iterations.

This also makes me think we could also add some circular dependency protection here.  I think if we check whether any of the `new_deps` already exist in the `vars_with_deps` would work.  Might be more informative than reaching the recursion limit error.